src/config.c

```diff
@@ -0,0 +1,197 @@
+                                           name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+
+    if (strcasecmp(name, "cf-bucket-size") == 0) {
+        rm_config.cf_initial_size.min = new_val * 2;
+    } else if (strcasecmp(name, "cf-initial-size") == 0) {
+        rm_config.cf_bucket_size.max = new_val / 2 < 255 ? new_val / 2 : 255;
+    }
+
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getNumericValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%zu", *((size_t *)privdata));
+}
+
+typedef struct {
+    const char *name;
+    const char *defVal;
+    RedisModuleConfigSetStringFunc setValue;
+    RedisModuleConfigGetStringFunc getValue;
+    void *privdata;
+} RM_ConfigVar;
+
+RM_ConfigVar rm_config_vars[] = {
+    {
+        .name = "bf-error-rate",
+        .defVal = "0.01",
+        .setValue = setFloatValue,
+        .getValue = getFloatValue,
+        .privdata = &rm_config.bf_error_rate,
+    },
+    {
+        .name = "bf-initial-size",
+        .defVal = "100",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_initial_size,
+    },
+    {
+        .name = "bf-expansion-factor",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_expansion_factor,
+    },
+    {
+        .name = "cf-bucket-size",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_bucket_size,
+    },
+    {
+        .name = "cf-initial-size",
+        .defVal = "1024",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_initial_size,
+    },
+    {
+        .name = "cf-max-iterations",
+        .defVal = "20",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_max_iterations,
+    },
+    {
+        .name = "cf-expansion-factor",
+        .defVal = "1",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_expansion_factor,
+    },
+    {
+        .name = "cf-max-expansions",
+        .defVal = "32",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_max_expansions,
+    },
+};
+
+int RegisterStringConfig(RedisModuleCtx *ctx, RM_ConfigVar var) {
+    return RedisModule_RegisterStringConfig(ctx, var.name, var.defVal,
+                                            REDISMODULE_CONFIG_UNPREFIXED, var.getValue,
+                                            var.setValue, NULL, var.privdata);
+}
+
+int RM_RegisterConfigs(RedisModuleCtx *ctx) {
+    RedisModule_Log(ctx, "notice", "Registering configuration options");
+
+    for (int i = 0; i < sizeof rm_config_vars / sizeof *rm_config_vars; ++i) {
```

Comment:
Can we please use `sizeof` with the function syntax? :-)
Also here - comparing the signed and unsigned integers within the same row. And later, using `i` declared as `int` as an index for the `rm_config_vars` which also requires an unsigned as the receiving type.

Reply:
sizeof with function syntax on an expression is :vomiting_face:

Reply:
arrays may be indexed by `int`s in C. an index array does not need to be an unsigned type in C. because C is special like that.

Reply:
Have I ever told you I hated C for stupid things like that? I wonder how they would address (and call it perfectly valid) an array with an index of `-100`.

Reply:
`arr[idx]` is simply sugar for `*(arr + idx)`. subtraction of an integral value from a pointer is valid in C as long as the resulting pointer obtained from the subtraction is a valid pointer to within the original array object.
hope that helps :sweat_smile:

Reply:
> `arr[idx]` is simply sugar for `*(arr + idx)`. subtraction of an integral value from a pointer is valid in C as long as the resulting pointer obtained from the subtraction is a valid pointer to within the original array object. hope that helps 😅

Yes, I read the K&R book a long time ago. This doesn't make C better. As well as iterating over an array till its end with a signed integer :-)